crud/crud.cpp

```diff
@@ -243,45 +277,28 @@ crud::handle_size(const bzn::message& msg, bzn::message& response)
     response["data"]["size"] = Json::UInt64(this->storage->get_size(msg["db-uuid"].asString()));
 }
 
-void
-crud::do_raft_task_routing(const bzn::message& msg, bzn::message& response)
-{
-    switch(raft->get_state())
-    {
-        case bzn::raft_state::candidate:
-            response = do_candidate_tasks(msg);
-            break;
-        case bzn::raft_state::follower:
-            response = do_follower_tasks(msg);
-            break;
-        case bzn::raft_state::leader:
-            response = do_leader_tasks(msg);
-            break;
-        default: // A bad RAFT State?!?!?! This should not happen
-            response["request-id"] = msg["request-id"];
-            response["error"] = bzn::MSG_INVALID_RAFT_STATE;
-            assert(false);
-            break;
-    }
-}
-
 void
 crud::handle_ws_crud_messages(const bzn::message& msg, std::shared_ptr<bzn::session_base> session)
 {
     LOG(debug) << msg.toStyledString();
 
     auto response = std::make_shared<bzn::message>();
+    (*response)["request-id"] = msg["request-id"];
 
-    assert(msg["bzn-api"]=="crud");
-
-    if( auto search = accepted_crud_commands.find(msg["cmd"].asString()); search != accepted_crud_commands.end() )
+    if( !(msg.isMember("bzn-api") && msg["bzn-api"]=="crud"))
     {
-        this->do_raft_task_routing(msg, *response);
+        (*response)["error"] = bzn::MSG_INVALID_CRUD_COMMAND;
     }
     else
     {
-        (*response)["request-id"] = msg["request-id"];
-        (*response)["error"] = bzn::MSG_INVALID_CRUD_COMMAND;
+        if( auto search = accepted_crud_commands.find(msg["cmd"].asString()); search != accepted_crud_commands.end() )
```

Comment:
search is not used... do:

if (this->accept_crud_commands.find() != this->accept_crud_commands.end())
...